mobidam-sst-management-frontend/frontend/src/composables/rules.ts

```diff
@@ -27,19 +27,48 @@ export function useRules() {
             (value && value.length <= length) || message;
     }
 
-    function notEmptyDateRule(message = "error") {
+    function notEmptyRule(message = "error") {
         return (value: string | null | undefined) =>
-            (value && value.trim() != "-") || message;
+            (value && value.trim() != "") || message;
     }
 
-    function notEmptyRule(message = "error") {
+    function isValidEmail(message = "error") {
+        const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
         return (value: string | null | undefined) =>
-            (value && value.trim() != "") || message;
+            (value && emailPattern.test(value.trim())) || message;
+    }
+
+    function isExpiredGueltigBis(toDate: string | undefined) {
```

Comment:
No need to use specifier, just name it: function isExpired(date: string | undefined) {

Reply:
✅